ports/glutin/window.rs

```diff
@@ -605,15 +607,18 @@ impl WindowMethods for Window {
     }
 
     fn set_page_title(&self, title: Option<String>) {
+        let current_url = self.current_url.borrow().clone().unwrap();
+
         let title = match title {
             Some(ref title) if title.len() > 0 => &**title,
-            _ => "untitled",
+            _ => &current_url,
         };
         let title = format!("{} - Servo", title);
         self.window.set_title(&title);
     }
 
-    fn set_page_url(&self, _: Url) {
+    fn set_page_url(&self, url: Url) {
```

Comment:
@KiChjang its called in `components/compositing/compositor.rs` at "change_page_url" method, which's executed in the same file:

```rust
(Msg::ChangePageTitle(pipeline_id, title), ShutdownState::NotShuttingDown) => {
    self.change_page_title(pipeline_id, title);
}

(Msg::ChangePageUrl(pipeline_id, url), ShutdownState::NotShuttingDown) => {
    self.change_page_url(pipeline_id, url);
}
```